.github/workflows/ci.yaml

```diff
@@ -0,0 +1,292 @@
+name: release harmony
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+  check:
+    name: Per-check for current tag
+    runs-on: ubuntu-18.04
+    continue-on-error: false
+    outputs:
+      tag_annotated: ${{ steps.check-tag-annotated.outputs.tag_annotated }}
+
+    steps:
+      - name: Checkout harmony core code
+        uses: actions/checkout@v2
+        with:
+          path: harmony
+          ref: main
+          fetch-depth: 0
```

Comment:
what does this 0 mean here?  we can actually do shallow checkout to speed up.

Reply:
set 0 means need fetch all history commit. we need to generate version automatically in next steps otherwise will only create v0-0 instead of correct version. and this change only take more few seconds than not fetch all history commit